SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,952 @@
+
+                if (!MethodValid)
+                {
+                    // Method not found, log error and skip.
+                    Logger.Error($"[ConfigModOptions] Could not find the specified method: {typeof(T)}.{Name}");
+                    return;
+                }
+
+                Traverse.Create(config).Method(Name, MethodParameterTypes).GetValue(arguments);
+            }
+        }
+
+        private struct ConfigModOptionsMetadata
+        {
+            /// <summary>
+            /// Timestamp of the relevant mod assembly. Used to determine the validity of the metadata.
+            /// </summary>
+            public long Timestamp;
+
+            /// <summary>
+            /// The <see cref="MenuAttribute"/> relating to this <see cref="ModOptions"/> menu.
+            /// </summary>
+            public MenuAttribute MenuAttribute;
+
+            /// <summary>
+            /// A dictionary of <see cref="ModOptionMetadata"/>, indexed by <see cref="ModOption.Id"/>.
+            /// </summary>
+            public Dictionary<string, ModOptionMetadata> ModOptionsMetadata;
+        }
+
+        private ConfigModOptionsMetadata configModOptionsMetadata;
+
+        private static readonly string CacheDir = Path.Combine(
+            Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location),
+            "OptionsPanelCache");
+        private static Dictionary<string, ConfigModOptionsMetadata> Cache;
+
+        private string CachePath => Path.Combine(CacheDir, $"{QMod.Id}.dat");
+        private string CacheKey => typeof(T).FullName;
+
+        private void LoadMetadata()
+        {
+            long timestamp = File.GetLastWriteTimeUtc(assembly.Location).Ticks;
+            Logger.Debug($"Timestamp: {timestamp}");
+
+            if (Cache == null)
+                LoadCache();
+
+            Logger.Debug($"OptionsPanelCache: {Cache?.Count() ?? 0} entries.");
+
+            if (Cache != null && Cache.TryGetValue(CacheKey, out var metadata) && timestamp == metadata.Timestamp)
+            {
+                Logger.Debug($"[{QMod.DisplayName}] {CacheKey}: Loaded metadata from OptionsPanelCache.");
+                configModOptionsMetadata = metadata;
+                Name = configModOptionsMetadata.MenuAttribute.Name;
+            }
+            else
+            {
+                Logger.Debug($"[{QMod.DisplayName}] {CacheKey}: Parsing metadata from attributes via reflection.");
+
+                configModOptionsMetadata = new ConfigModOptionsMetadata
+                {
+                    MenuAttribute = GetMenuAttributeOrDefault(),
+                    ModOptionsMetadata = new Dictionary<string, ModOptionMetadata>(),
+                    Timestamp = timestamp
+                };
+                Name = configModOptionsMetadata.MenuAttribute.Name;
+
+                ProcessMetadata();
+                SaveCache();
+            }
+        }
+
+        private void LoadCache()
```

Comment:
Tbh, I have wondered this. I mostly implemented it bc @PrimeSonic was concerned about LINQ+Reflection being too heavy, but IMO this caching approach is likely overzealous optimisation that might not actually be the "optimal" way anyway.

We could consider implementing a stopwatch to time how long each approach takes, but until we have numbers to work with its kind of impossible to know whether this serialized cache is worth it. It might turn out to be a worthwhile thing to turn on later, if users begin to report slow startup times.

Perhaps we could even make it an option. "Is startup a little slow for you? Try turning on the cache."

Reply:
I'm using profilers often with my reflection/linq/harmony stuff. LINQ/reflection can be slow, but I think it needed to be really heavy LINQ stuff to be even near disk operations. I mean, simple logging in the middle can change numbers a lot.

Reply:
On my machine using config with around 40 mixed members:
```
[SMLHelper/Info] LoadCache: 50.385 ms
[SMLHelper/Info] ProcessMetadata: 6.6284 ms
[SMLHelper/Info] SaveCache: 20.2819 ms
```
Also:
`[SMLHelper/Info] BindEvents: 0.6095 ms` - with LINQ checks
`[SMLHelper/Info] BindEvents: 0.1359 ms` - without checks

Reply:
Worse case scenario, we add a couple seconds to the initial game load time.  
And from what we are seeing in Experimental, that is going to be going up significantly regardless of mods.  
So I will be easing off on any performance concerns, especially if it involves code that only runs once at start anyways.

Reply:
Yeah, I did think it might be the case that in many scenarios, the IO is going to be slower. Thank you for taking the time to confirm that.

As for the with/without LINQ checks for bind events, you might be onto something. I would prefer to also test it with alternative, non-LINQ checks that I have yet to write.

Reply:
Resolved by 4596875